custom_components/opnsense/repair_reconciliation.py

```
@@ -0,0 +1,274 @@
+                )
+            migrations.append((candidate, target_unique_id))
+
+        old_main = device_registry.async_get_device(
+            identifiers={(DOMAIN, self.marker.old_device_id)}
+        )
+        new_main = device_registry.async_get_device(
+            identifiers={(DOMAIN, self.marker.new_device_id)}
+        )
+        if new_main is not None:
+            if old_main is not None and new_main.id != old_main.id:
+                raise RepairReconciliationError("primary device identifier target collision")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)